### DISCUSSION TITLE
Best way to deploy the psappdeploytoolkit v4.0 module to clients?

### DISCUSSION THREAD

**Discussion by sthar13** [PSAppDeployToolkit/PSAppDeployToolkit#1976] [Ideas]
### Summary of the new feature / enhancement

Hello,

What would the best way to deploy and maintain the ps module v4.0 to all clients and keep it maintained? Best not to use the online installer with install-module. What would be your approach?

**Comment by DanGough** [comment]
Simplest approach would be to just bundle it with each deployment package (as it is when you create a template), then it does not need to be pre-installed on the clients.

Otherwise, it would need to be packaged. If there is sufficient demand for this, we could provide an MSI. Chocolatey and Winget packages are also a possibility.

**Comment by sthar13** [comment]
Well we have a lot of deployments. I would not want the ccmcache to pile up with multiple copies of psappdeploytoolkit module. So, for me an msi package would be the solution. I can always copy it to psmodules folder of users' PCs via scripting but a universal community approved approach to this would be awesome.

**Comment by DanGough** [comment]
This could be done...

![image](https://github.com/user-attachments/assets/d8f42369-0d5c-44fc-8a8b-a9caf46cced9)

We'd have to discuss internally to see if we want to support alternative installation methods however.

**Comment by noyard** [comment]
I would like Winget if possible, it is built into windows 11 and PSADT could be updated with just one command line.

**Comment by That-Annoying-Guy** [comment]
I would never want to install the PSADT module on clients.
When you deploy a package with a known version of PSADT, you know it works.
When you change the installed version of the module on a client, you might affect how the package install.
Will an old package install with a new module? Will the old package uninstall with the new module?

The PSADT module evolves with time. unless you rebuild clients every few month without fail, old packages could break with newer versions of the module. The behavior of functions could change with time. Something works on day one and a year later it breaks because of an unknown race condition. Parameters and functions could be renamed or removed.

PSADT is shrinking with each release. 
The CCMCACHE is also a rolling cache. After installation, the cache is eventually replaced or emptied.
